app/integrations/zendesk/scheduler.py

```
@@ -5540,6 +5540,35 @@ async def _process_window(
+            reason: str | None = None,
+            error: str | None = None,
+        ) -> None:
+            nonlocal note_logged
+            if note_logged:
+                return
+            note_logged = True
+            posted_label = "yes" if posted else "no"
+            if error:
+                logger.info(
+                    "zendesk_internal_note_result ticket_id=%s posted=%s reason=%s dry_run=%s error=%s",
+                    tid,
```

Ruff (0.14.14)

[warning] Function definition does not bind loop variable `tid`

(B023)